### PR TITLE
exp/services/recoverysigner: update API to reflect SEP-30 v0.2.0

### PR DESCRIPTION
**Pull request by leighmcculloch**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Update API and related code to reflect the changes proposed in SEP-30 v0.2.0 in stellar/stellar-protocol#566.

### Why

To align the reference implementation with the SEP change being made to allow for easier expansion in the future of both additional roles and multiple values for each identity type (email, phone, etc).

I'm updating this code before finalizing the protocol change because I wanted to validate the protocol change in code. I'll merge both together.

### CODE REVIEW

File: exp/services/recoverysigner/internal/serve/account_post.go [hunk 1]
```diff
@@ -57,18 +56,22 @@ func (h accountPostHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		Address:    req.Address.Address(),
+		Identities: []account.Identity{},
+	}
+	for _, i := range req.Identities {
+		accIdentity := account.Identity{
+			Role: i.Role,
+		}
+		for _, m := range i.AuthMethods {
+			accIdentity.AuthMethods = append(accIdentity.AuthMethods, account.AuthMethod{
+				Type:  account.AuthMethodType(m.Type), // TODO: Validate AuthMethodType
```
Comment (accordeiro) [inline exp/services/recoverysigner/internal/serve/account_post.go]: Do we need to be opinionated about the AuthMethod types? Or are we always simply going to compare the stored `method` with the one provided in the JWT?

Comment (accordeiro) [reply exp/services/recoverysigner/internal/serve/account_post.go]: After reading the rest of the code, it seems like we are explicit all over the codebase about the methods we support. In that case, shouldn't we add this validation now?

Comment (leighmcculloch) [reply exp/services/recoverysigner/internal/serve/account_post.go]: >are we always simply going to compare the stored method with the one provided in the JWT?

This is an interesting idea, but I could see it being easy to make a mistake leading to an exploit. I'd rather be explicit. Any implementer can add new auth methods and support them too.

We can add validation now. I'll add it before merging.

File: exp/services/recoverysigner/internal/account/account.go [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
+
+type Identity struct {
+	Role        string
+	AuthMethods []AuthMethod
+}
+
+type AuthMethodType string
+
+const (
+	AuthMethodTypeAddress     AuthMethodType = "stellar_address"
```
Comment (howardtw) [inline exp/services/recoverysigner/internal/account/account.go]: I find this name a little confusing while I'm reviewing the code as I'm not sure whether it's a stellar address or an email address.

Comment (leighmcculloch) [reply exp/services/recoverysigner/internal/account/account.go]: Inside the application wherever `address` is referenced it means stellar address. I think everything will get a bit wordy if we add the word `stellar` prefixed too. If you look a couple lines down email is right there as another option, so I doubt too many people will make this mistake?

**Comment by leighmcculloch** [comment]
>I think it's time to add a SPEC for the service.

I don't plan to write a `SPEC.md` file for this service. SEP-30 is the specification. If anything isn't clear it should be added there, not here. This is just a reference implementation of SEP-30. Can you open issues or PRs for anything in SEP-30 that isn't clear? Anything you already mentioned on PRs I've got noted down to add, so no need to open anything that's already been mentioned.

**Pull request merged**